packages/pwa/app/utils/utils.js

```diff
@@ -162,3 +163,153 @@ export const capitalize = (text) => {
  * @returns {object} - url object from the pwa-kit-config.json file
  */
 export const getUrlConfig = () => pwaKitConfig?.app?.url
+/**
+ * A util to return current multi site configuration
+ * @returns {object} - a list of site objects from the pwa-kit-config.json file
+ */
+export const getSitesConfig = () => pwaKitConfig?.app?.sites
+/**
+ * A util to return current defaultSiteId
+ * @returns {string} - default site Id string from pwa-kit-config file
+ */
+export const getDefaultSiteId = () => pwaKitConfig?.app?.defaultSiteId
+
+/**
+ * Return a the configuration of the given key from pwa-kit-config
+ * By default, returns app configuration
+ * @param keys {array} - array of keys
+ * @returns {object}
+ */
+export const getConfig = () => pwaKitConfig
+
+export const getSite = (url) => {
+    const {
+        app: {defaultSiteId, sites: sitesConfig}
+    } = getConfig()
+    let path = url
+    if (!path) {
+        path = `${window?.location.pathname}${window?.location.search}`
+    }
+    let site
+    site = getSiteByUrl(path)
+    if (site) {
+        return site
+    }
+
+    const {hostname} = new URL(`${getAppOrigin()}${path}`)
+    site = getSiteByHostname(hostname)
+
+    if (site) {
+        return site
+    }
+
+    site = sitesConfig.find((site) => site.id === defaultSiteId)
+
+    return site
+}
+
+/**
+ * Get the site Id based on the given hostname
+ * @param {string} hostname
+ * @returns {string} siteId
+ */
+export const getSiteByHostname = (hostname) => {
+    const {
+        app: {sites: sitesConfig}
+    } = getConfig()
+    if (!sitesConfig.length) throw new Error('No site config found. Please check you configuration')
+    if (!hostname) return undefined
+
+    const site = sitesConfig.filter((site) => {
+        return site?.hostnames?.some((i) => i.includes(hostname))
+    })
+
+    return site?.length === 1 ? site[0] : undefined
```

Comment:
Just an idea, but what are your thoughts on using a 3rd party library like [jsonpath](https://www.npmjs.com/package/jsonpath). It would allow you to do something like this use JSONPath expressions to find the site by crafting a path, instead of manually looping/filtering.

Reply:
Thanks for the suggestion, I had a look on the `jsonpath`, the application might look handy, but it also introduces some conventions/syntax that require some reading/comprehension. That means when a dev reads the code, they have to spend some time studying the package. Is it worth doing so? 🤔 

Unless our configuration object is complicated with lots of nested data, I find that using js filtering and looping is relatively more readable than using `jsonpath`.